src/lib/ndpi_main.c

```diff
@@ -126,7 +129,7 @@ void ndpi_flow_free(void *ptr) {
 /* ****************************************** */
 
 void *ndpi_realloc(void *ptr, size_t old_size, size_t new_size) {
-  void *ret = ndpi_malloc(new_size);
+  void *ret = ndpi_calloc(new_size, sizeof(char));
```

Comment:
I would discourage to alter the behavior of `ndpi_realloc`. IMHO it should behave like a libc `realloc` which does not zero any memory.

Reply:
I strongly suggest you chase down the real bugs and fix them. Changing the semantic of the realloc operation is not a good idea, IMO
My 2 cents